Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+
+For e.g.:
+
+```java
+ 
+ //assuming vehicle is an instance of Class `Car` the expression inside the 'if' will  return true
+ if( vehicle instanceof Car ) {
+   //do something if vehicle is a Car
+ }
+```
+
+## Control Flow
+
+Control flow statements help a developer to take `decisions` based on values. Primarily, Java has the following constructs for control flow:
+
+* `if`
+
+```java
+        if( <expression that results in a boolean>) {
+         //code enters this block if the above expression is 'true'
+        }
+```
+
+* `if...else`
+   
+```java
+        if( <expression that results in a boolean> ){
+          //execute this block if the expression is 'true'
+        
+        } else{
+          //execute this block if the expression is 'false'
+        }
+```
+
+* `switch`
+
+Switch is an alternative for the `if...else` construct when there are multiple values and cases to check against. 
+
+```java
+        switch( <integer / String / Enum > ){
+            case <int/String/Enum> : <statements>
+            break;
+            case <int/String/Enum> : <statements>
+            default: <statements>
+        }
+```
+ 
+ Note: The program flow `falls through` the next `case` if the `break` statement is missing. For e.g. Let's say you say the standard 'Hello' to everyone at office, but you are extra nice to the girl who sits next to you and sound grumpy to your boss. The way to represent would be something like:
+    
+```java
+    switch(person){
+           case 'boss' : soundGrumpy();
+           break;
+           case 'neighbour' : soundExtraNice();
+           case 'colleague': soundNormal();
+           break;
+           default :
+           soundNormal();
+       }
+```
+    Note: The `default` case runs when none of the `case` matches. Remember that when a case has no `break` statement, it `falls through` to the next case and will continue to the subsequent `cases` till a `break` is encountered.
+    
+* `nested statements`
+
+Any of the previous control flows can be nested. Which means you can have nested `if`,`if..else` and `switch..case` statements. i.e., you can have any combination of these statements within the other and there is no limitation to the depth of `nesting`.
+    
+    
+## Loop constructs
+
+
+Remember the guy (or gal) who goes on and on about something? Well, that's a `loop` right there for you. In `Java`, a developer can represent the action of  `doing something repeatedly` using the following constructs:
+
+###### `while` Loop
+
+
+The `while` loop executes a group of statements / single statement till a condition evaluates to `true`. Like so:
+
+```java
+ while( some condition is true ){
+  //do something
+  //update the condition
+ }
+```
+`Note`: For the `while` loop to start executing, you'd require the condition to be true. To exit the loop, you'd require to modify the condition so that it becomes false at some point of time. Otherwise, the loop will continue to execute forever.
```

Comment:
Once the loop starts executing, there is no modification!

Rewrite the sentence in a way, that it is clear.